cupyx/optimizing/_optimize.py

```diff
@@ -83,7 +90,18 @@ def optimize(*, key=None, **config_dict):
     context = _optimize_config.get_new_context(key, _optimize, config_dict)
     _optimize_config.set_current_context(context)
 
+    if path is not None:
+        if os.path.exists(path):
+            context.load(path)
+        elif readonly:
+            warnings.warn('''
+The specified path {} could not be found, and the readonly option is set.
+The optimization results will never be stored.
+                '''.format(path))
```

Comment:
Remove trailing spaces from the warning message.
```suggestion
'''.format(path))
```